lib/core/src/Cardano/Wallet/Primitive/AddressDiscovery/Shared.hs

```diff
@@ -305,17 +314,37 @@ accountXPubCondition
 accountXPubCondition accXPub (ScriptTemplate cosignerKeys _) =
     getRawKey accXPub `F.elem` cosignerKeys
 
-walletCreationInvariant
+validateScriptTemplates
     :: WalletKey k
     => k 'AccountK XPub
+    -> ValidationLevel
     -> ScriptTemplate
     -> Maybe ScriptTemplate
-    -> Bool
-walletCreationInvariant accXPub pTemplate dTemplate =
-    isValid pTemplate && maybe True isValid dTemplate
+    -> Either (CredentialType, Text) ()
```

Comment:
It would be nice to wrap this return code up in something like
```haskell
data ErrScriptTemplateInvalid = ErrScriptTemplateInvalid !CredentialType !ErrValidateScriptTemplate
instance ToText ErrScriptTemplateInvalid where
    -- ...
```

Reply:
done